src/planner/parabolic/HauserParabolicSmootherHelpers.cpp

```diff
@@ -0,0 +1,198 @@
+#include "HauserParabolicSmootherHelpers.hpp"
+#include <chrono>
+#include <cmath>
+#include <aikido/util/VanDerCorput.hpp>
+#include "Config.h"
+#include "HauserMath.h"
+#include "ParabolicUtil.hpp"
+
+using namespace ParabolicRamp;
+
+namespace aikido {
+namespace planner {
+namespace parabolic {
+namespace detail {
+
+class SmootherFeasibilityCheckerBase
+    : public ParabolicRamp::FeasibilityCheckerBase
+{
+public:
+  SmootherFeasibilityCheckerBase(
+      aikido::constraint::TestablePtr testable, double checkResolution)
+    : mTestable(std::move(testable))
+    , mCheckResolution(checkResolution)
+    , mStateSpace(mTestable->getStateSpace())
+    , mInterpolator(mStateSpace)
+  {
+    // Do nothing
+  }
+
+  bool ConfigFeasible(const ParabolicRamp::Vector& x) override
+  {
+    Eigen::VectorXd eigX = toEigen(x);
+    auto state = mStateSpace->createState();
+    mStateSpace->expMap(eigX, state);
+    return mTestable->isSatisfied(state);
+  }
+
+  bool SegmentFeasible(
+      const ParabolicRamp::Vector& a, const ParabolicRamp::Vector& b) override
+  {
+    Eigen::VectorXd eigA = toEigen(a);
+    Eigen::VectorXd eigB = toEigen(b);
+
+    auto testState = mStateSpace->createState();
+    auto startState = mStateSpace->createState();
+    auto goalState = mStateSpace->createState();
+    mStateSpace->expMap(eigA, startState);
+    mStateSpace->expMap(eigB, goalState);
+
+    aikido::util::VanDerCorput vdc{1, false, false, mCheckResolution};
```

Comment:
Nit: Add a comment explaining why `includeStartpoint` and `includeEndpoint` can be `false`.